docs/datastore/datastore-api.md

```diff
@@ -0,0 +1,32 @@
+---
```

Comment:
IMHO this should be generated from typedoc. We can keep this but it will outdate really fast.

Reply:
Can we? I know users could generate from javadoc in Java. I will check for tools to help with this

Reply:
Do not check/do stuff without issue. There are tools like typedoc. We can do that in no time. 
Just do not do stuff manually.